Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/SendSearchRequestToDrivers/Handle/Internal/DriverPool.hs

```diff
@@ -186,36 +187,36 @@ prepareDriverPoolBatch driverPoolCfg searchReq searchTry batchNum goHomeConfig =
               driverPoolCfg.distanceBasedBatchSplit
 
         calcGoHomeDriverPool = do
-          calculateGoHomeDriverPool $
-            CalculateGoHomeDriverPoolReq
-              { poolStage = DriverSelection,
-                driverPoolCfg = driverPoolCfg,
-                goHomeCfg = goHomeConfig,
-                variant = Just searchTry.vehicleVariant,
-                fromLocation = searchReq.fromLocation,
-                toLocation = searchReq.toLocation, -- last or all ?
-                merchantId = searchReq.providerId
-              }
-
+          case searchReq.tag of
+            DSR.ON_DEMAND ->
+              calculateGoHomeDriverPool $
+                CalculateGoHomeDriverPoolReq
+                  { poolStage = DriverSelection,
+                    driverPoolCfg = driverPoolCfg,
+                    goHomeCfg = goHomeConfig,
+                    variant = Just vehicleVariant,
+                    fromLocation = searchReq.searchRequestDetails.fromLocation,
+                    toLocation = searchReq.searchRequestDetails.toLocation,
+                    merchantId = searchReq.providerId
+                  }
+            DSR.RENTAL -> pure [] -- RENTAL
         calcDriverPool radiusStep = do
-          let vehicleVariant = searchTry.vehicleVariant
-              merchantId = searchReq.providerId
-          let pickupLoc = searchReq.fromLocation
+          let merchantId = searchReq.providerId
+          let pickupLoc = searchReq.searchRequestDetails.fromLocation
```

Comment:
Failed for rental case, because `fromLocation` field isn't there for rentals. Working on removing this unsafe code